File: Extension Objects/Runtime Ports/_Source Codes (iOS & MacOS)/iOS/CRunOpenURLs.m
#import "CRunOpenURLs.h"
#import "CFile.h"
#import "CRunApp.h"
#import "CBitmap.h"
#import "CCreateObjectInfo.h"
#import "CValue.h"
#import "CExtension.h"
#import "CRun.h"
#import "CCndExtension.h"
#import "CActExtension.h"

// ~~~~~~~ DEFINE A/C/Es HERE ~~~~~~~
#define CND_LAST 0

#define ACT_OPENURL 0

@implementation CRunOpenURLs

-(int)getNumberOfConditions
{
    return CND_LAST;
}

-(BOOL)createRunObject:(CFile*)file withCOB:(CCreateObjectInfo*)cob andVersion:(int)version
{
    return YES;
}

-(void)destroyRunObject:(BOOL)bFast
{
    
}

-(int)handleRunObject
{
    return REFLAG_ONESHOT;
}

-(void)displayRunObject:(CRenderer*)renderer
{
    
}

// ~~~~~~~ CONDITIONS ~~~~~~~

-(BOOL)condition:(int)num withCndExtension:(CCndExtension *)cnd
{
    return NO;
}

// ~~~~~~~ ACTIONS ~~~~~~~

-(void)action:(int)num withActExtension:(CActExtension*)act
{
    switch(num)
    {
        case ACT_OPENURL:
            NSString* url=[act getParamExpString:rh withNum:0];
            [[UIApplication sharedApplication] openURL:[NSURL URLWithString:url]];
            break;
    }
}

// ~~~~~~~ EXPRESSIONS ~~~~~~~

// NaN

@end

File: Extension Objects/Runtime Ports/_Source Codes (XNA & Java)/XNA/CRunOpenURLs.cs
// OpenURLs - XNA port (Windows & Windows Phone)

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using RuntimeXNA.Extensions;
using RuntimeXNA.Services;
using RuntimeXNA.RunLoop;
using RuntimeXNA.Sprites;
using RuntimeXNA.Conditions;
using RuntimeXNA.Actions;
using RuntimeXNA.Expressions;
using RuntimeXNA.Objects;
using RuntimeXNA.Params;
using RuntimeXNA.Frame;
using RuntimeXNA.OI;
using RuntimeXNA.Movements;

namespace RuntimeXNA.Extensions
{
    class CRunOpenURLs : CRunExtension
    {
        //const int CND_LAST = 0;
        
        const int ACT_OPENURL = 0;

        public override int getNumberOfConditions()
        {
            return 0;
        }

        public override bool createRunObject(CFile file, CCreateObjectInfo cob, int version)
        {
            return true;
        }

        public override bool condition(int num, CCndExtension cnd)
        {
            return false;
        }

        public override void action(int num, CActExtension act)
        {
            switch (num)
            {
                case ACT_OPENURL:
                {
                    string url = act.getParamExpression(rh, 0);

                    #if WINDOWS
                        Process.Start(url);
                    #elif WINDOWS_PHONE
                        WebBrowserTask browser = new WebBrowserTask();
                        browser.Uri = new Uri(url);
                        browser.Show();
                    #elif XBOX
                        // Xbox-specific implementation. (Idk how it would be implemented on Xbox.)
                    #endif
                    break;
                }
            }
        }

        public override CValue expression(int num)
        {
            return new CValue(0);
        }
    }
}
